01-supp-addressingData.Rmd

```diff
@@ -0,0 +1,174 @@
+
+> ## Challenge - Selecting values 1 {.challenge}
+>
+> What will be returned by `dat[1,1]`?
+
+```{r indexing1}
+dat[1,1]
+```
+
+If we leave out a dimension R will interpret this as a request for all values in that dimension.
+
+> ## Challenge - Selecting values 2 {.challenge}
+>
+> What will be returned by `dat[,2]`?
+
+The colon `:` can be used to create a sequence of integers.
+
+```{r colonOperator}
+6:9
+```
+
+Creates a vector of numbers from 6 to 9.
+
+This can be very useful for addressing data. 
+
+> ## Challenge - Subsetting with sequences {.challenge}
+> Use the colon operator to index just the aneurism count data (columns 6 to 9).
+
+Finally we can use the `c()` (combine) function to address non-sequential rows and columns.
+
+```{r nonsequential_indexing}
+dat[c(1,5,7,9),1:5]
+```
+
+Returns the first 5 columns for patients in rows 1,5,7 & 9
+
+> ## Challenge - Subsetting non-sequential data {.challenge}
+> Return the Age and Gender values for the first 5 patients.
+
+### Addressing by name
+Columns in an R data frame are named.
+
+```{r column_names}
+names(dat)
+```
+
+> ## Tip {.callout} 
+>
+> If names are not specified e.g. using `headers=FALSE` in a `read.csv()` function, R assigns default names `V1,V2,...,Vn`
+
+We usually use the `$` operator to address a column by name
+
+```{r named_addressing}
+dat$Gender
+```
+
+Named addressing can also be used in square brackets.
+```{r names_addressing_2}
+head(dat[,c('Age','Gender')])
+```
+
+> ## Best Practice {.callout} 
+>
+> Best practice is to address columns by name, often you will create or delete columns and the column position will change.
+
+
+### Logical indexing
+
+A logical vector contains only the special values `TRUE` & `FALSE`.
+
+```{r logical_vectors}
+c(TRUE,TRUE,FALSE,FALSE,TRUE)
+```
+> ## Tip {.callout} 
+>
+> Note the values `TRUE` and `FALSE` are all capital letters and are not quoted.
+
+Logical vectors can be created using `relational operators` e.g. `<, >, ==, !=, %in%`.
+
+```{r logical_vectors_example}
+x<-c(1,2,3,11,12,13)
+x < 10
+
+x %in% 1:10
+```
+
+We can use logical vectors to select data from a data frame.
+
+```{r logical_vectors_indexing}
+index <- dat$Group == 'Control'
+dat[index,]$BloodPressure
+```
+
+Often this operation is written as one line of code:
+
+```{r logical_vectors_indexing2}
+plot(dat[dat$Group=='Control',]$BloodPressure)
+```
+
+> ## Challenge - Using logical indexes {.challenge}
+> 1. Create a scatterplot showing BloodPressure for subjects not in the control group.
+> 2. How many ways are there to index this set of subjects?
+
+### Combining indexing and assignment
+
+The assignment operator `<-` can be combined with indexing.
+
+```{r indexing and assignment}
+x<-c(1,2,3,11,12,13)
+x[x < 10] <- 0
+x
+```
+
+> ## Challenge - Updating a subset of values {.challenge}
+> In this dataset, values for Gender have been recorded as both uppercase `M, F` and lowercase `m,f`. 
```

Comment:
While the task is the same (update values from 'M' to 'm') the solution changes in PR #57 . The change from a character vector to a factor introduces a new step. The key new command in PR #57 is `droplevels()`.

Btw. Both lessons are designed to stand-alone. Personally, I would expect to teach addressing data before factors, but this is not a requirement.